src/hu_stuff.c

```diff
@@ -834,10 +833,10 @@ static void HU_widget_build_health (void)
   hud_healthstr[i] = '\0';
 
   // build the numeric amount init string
-  sprintf(hud_healthstr + i, "%3d", health);
+  sprintf(hud_healthstr + i, "%3d", st_health);
 
   // set the display color from the amount of health posessed
-  w_health.cr = ColorByHealth(health, 100, hu_invul);
+  w_health.cr = ColorByHealth(plr->health, 100, hu_invul);
```

Comment:
This is done intentionally, both for the Health and Armor widgets; having the widgets be colored based on the logical values gives you a rough idea of how low your health/armor is without having to wait for them to count all the way there (which doesn't take too long anyways).

Reply:
So, health widgets are colored by *target* health, and not the one currently displayed? Makes sense.

Reply:
That's another way to put it, yes.